### PR TITLE
Feature/aut 1195/e2e add asset passage to interaction

### PR DESCRIPTION
**Pull request by Silvia-loza**
**End to end testing.**
Related to: https://oat-sa.atlassian.net/browse/AUT-1195
Depends on :

- [oat-sa/tao-mediamanager#359](https://github.com/oat-sa/extension-tao-mediamanager/pull/359)
- https://github.com/oat-sa/tao-core/pull/3250

Description of changes:
Support functions (resource manager) for handling e2e tests for: Creating and importing passages/asset and adding to prompt and choice of interaction

How to run locally:
Checkout to branch `feature/AUT-1195/e2e-add-asset-passage-to-interaction`
in extensions:
- https://github.com/oat-sa/tao-core
- https://github.com/oat-sa/extension-tao-itemqti
- https://github.com/oat-sa/extension-tao-mediamanager

**run tests from tao core extn:*
npm run cy:open - to run in browser
or npm run cy:run - for headless execution
File to test: add-asset-passage-to-interaction.spec.js
**All tests should pass:**
image

### CODE REVIEW

**Review by adriloma — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following conventional commits
- [x] Commits messages are meaningful
- [x] Commits are atomic

File: views/cypress/utils/resource-manager.js [hunk 1]
```diff
@@ -43,3 +43,56 @@ export function selectUploadLocalAsset(fileName, pathToFile) {
             cy.getSettled(`#${resourcemgrId} li[data-file="/${fileName}"] .actions a.select`).click();
         });
 }
+/**
+ * Add/upload shared stimulus previously created in assets
+ * * @param {boolean} isCreatedAsset determines whether the asset is
+ * the one previously created (vs imported)
+ */
+export function selectUploadSharedStimulus(isCreatedAsset) {
```
Comment (bziondik) [inline views/cypress/utils/resource-manager.js:51]: This is related to taoMediaManager should be put in https://github.com/oat-sa/extension-tao-mediamanager/blob/master/views/cypress/utils/resource-manager.js
taoQtiItem can be used without taoMediaManager

File: views/cypress/utils/resource-manager.js [hunk 2]
```diff
@@ -43,3 +43,56 @@ export function selectUploadLocalAsset(fileName, pathToFile) {
...
+    return cy.get('.resourcemgr.modal')
+        .last()
+        .then(resourcemgr => {
+            const resourcemgrId = resourcemgr[0].id;
+            cy.getSettled(`#${resourcemgrId} .file-browser .root-folder`).should('exist');
+            cy.getSettled(`.mediamanager .folders .root`).should('exist');
+            cy.getSettled(`#${resourcemgrId} .mediamanager .folders .root ul > li`)
+                .first()
+                .click();
+            cy.getSettled(`.file-selector .files [data-alt="passage NEW.xml"]`).should('exist');
```
Comment (bziondik) [inline views/cypress/utils/resource-manager.js:62]: better to set name as param

**Review by bziondik — CHANGES_REQUESTED** [review]

**Review by bziondik — APPROVED** [review]
- [ ]  New code is covered by tests (if applicable)
- [ ]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [x]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful
image

**Pull request merged**